data_safe_haven/infrastructure/programs/sre/networking.py

```diff
@@ -1856,6 +1858,13 @@ def __init__(
             opts=child_opts,
             tags=child_tags,
         )
+        shm_provider = provider.Provider(
+            "shm_provider",
+            provider.ProviderArgs(
+                location=props.location,
```

Comment:
Is it always the case that the SHM and SRE will be deployed in the same region? Do we want/need this to be the case? If not, let's read the `shm_location` from the SHM config file.

Reply:
Added in d327acd